### PR TITLE
[DEV-3001] Update on-demand feature view function to conditionally fill null value

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR
* implements null filling value extractor to extract value to fill null value in the output column;
* updates on-demand feature view to include statement to fill null value if null filling operation is detected in the query graph.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/transform/null_filling_value.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        # inject null value to the function to extract null filling value
+        input_params = ", ".join(input_values)
+        null_filling_value_codes += f"\n\nfill_value = extract_null_filling_value({input_params})\n"
+        state.codes = null_filling_value_codes
+        scope: Dict[str, Any] = {}
+        exec(null_filling_value_codes, scope)  # pylint: disable=exec-used  # nosec
+        fill_value = scope["fill_value"]
+
+        # if fill_value is not null, that implies the query graph has null filling operation
```
Comment (kchua78) [inline featurebyte/query_graph/transform/null_filling_value.py:153]: If fill value is na should we skip the on-demand feature view altogether?

Comment (gee-senbong) [reply featurebyte/query_graph/transform/null_filling_value.py:153]: Nope, this class is used to extract fillna value, it is not used to construct on-demand feature view directly.

File: featurebyte/models/offline_store_ingest_query.py [hunk 2]
```diff
@@ -352,6 +358,7 @@ def initialize(
             self.odfv_info = odfv_info
 
         if self.is_decomposed:
+            # FIXME: should handle the case for null_filling_value
```
Comment (gee-senbong) [inline featurebyte/models/offline_store_ingest_query.py:361]: For DataBricks UDF, it is not handled in this PR.

File: featurebyte/query_graph/node/generic.py [hunk 3]
```diff
@@ -2076,6 +2075,12 @@ def _derive_user_defined_function_code(
...
-        statement = StatementStr(f"{var_name} = {value} if {flag_var_name} else {var_name}")
-        statements.append(statement)
+        expr_statements, var_name = self._convert_expression_to_variable(
+            var_name_expression=ExpressionStr(f"{value} if {flag_var_name} else {var_name}"),
+            var_name_generator=var_name_generator,
+            node_output_type=NodeOutputType.SERIES,
+            node_output_category=NodeOutputCategory.FEATURE,
+            to_associate_with_node_name=True,
+        )
+        statements.extend(expr_statements)
```
Comment (gee-senbong) [inline featurebyte/query_graph/node/generic.py:2085]: Update this part to avoid reassign the `var` with other values.

**Review by YS-L — APPROVED** [review]

**Pull request merged**